### PR TITLE
feat(issues): add SmartlingIssuesApi covering 18 endpoints

### PR DESCRIPTION
**Pull request by PavelLoparev**
## Summary

Adds `SmartlingIssuesApi` to the SDK, covering all 18 endpoints requested under the Issues v2 API (create / edit / open-or-close / get details / attribute updates / comments CRUD / find / count for both project and account scopes).

- 7 new enums (`IssueType`, `IssueSubType`, `IssueSeverityLevel`, `IssueState`, `IssueWatchingState`, `JobFilterPresence`, `IssueSortField`)
- 14 DTO interfaces and 14 parameter classes (flat-hierarchy generic `BaseIssuesFilterParameters<T>` backs the four find/count classes)
- Setter-side validation (`SmartlingException`) for text length, array limits, and non-negative limit/offset — matches the tags pattern
- 34 new unit tests across `Methods` (18), `Parameter validation` (11), and `Edge cases` (5); full project suite 282 passing
- Root re-exports added so consumers can import directly from the package

Quirks worth flagging in review:
- `updateIssueComment` uses **POST** (not PUT) per the official Smartling swagger — inline comment explains it
- `FindProjectIssuesParameters`/`FindAccountIssuesParameters` extend the base directly rather than their `Count*` counterparts; F-bounded self-typing would otherwise break fluent chains
- Design spec at `docs/superpowers/specs/2026-05-12-issues-api-client-design.md`, implementation plan at `docs/superpowers/plans/2026-05-12-issues-api-client.md`

## Test plan

- [x] `npm run build:dev` succeeds
- [x] `npm run pretest` clean for new files (`api/issues/**`, `test/issues.spec.ts`, `index.ts`)
- [x] `npm run test:dev` — 282 passing, 0 failing
- [x] Reviewer: verify URL/verb mapping in `api/issues/index.ts` against `docs/superpowers/specs/2026-05-12-issues-api-client-design.md` § "Endpoint mapping"
- [ ] Reviewer: confirm POST-for-`updateIssueComment` aligns with the production API's expectation

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: api/issues/params/base-issues-filter-parameters.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        if (jobFilter && jobFilter.jobUids && jobFilter.jobUids.length > 1000) {
+            throw new SmartlingException("Job filter jobUids array must not exceed 1000 items");
+        }
+        this.set("jobFilter", jobFilter);
+        return this as unknown as T;
+    }
+
+    setStringFilter(stringFilter: StringFilterDto): T {
+        this.set("stringFilter", stringFilter);
+        return this as unknown as T;
```
Comment (vsolovei-smartling) [inline api/issues/params/base-issues-filter-parameters.ts]: This is unnecessarily complex
```suggestion
export abstract class BaseIssuesFilterParameters extends BaseParameters {
    setCreatedDateBefore(date: string): this {
        this.set("createdDateBefore", date);
        return this;
    }

    setCreatedDateAfter(date: string): this {
        this.set("createdDateAfter", date);
        return this;
    }

    setResolvedDateBefore(date: string): this {
        this.set("resolvedDateBefore", date);
        return this;
    }

    setResolvedDateAfter(date: string): this {
        this.set("resolvedDateAfter", date);
        return this;
    }

    setAnswered(answered: boolean): this {
        this.set("answered", answered);
        return this;
    }

    setReopened(reopened: boolean): this {
        this.set("reopened", reopened);
        return this;
    }

    setHasComments(hasComments: boolean): this {
        this.set("hasComments", hasComments);
        return this;
    }

    setIssueSeverityLevelCodes(codes: Array<IssueSeverityLevel>): this {
        this.set("issueSeverityLevelCodes", codes);
        return this;
    }

    setIssueStateCodes(codes: Array<IssueState>): this {
        this.set("issueStateCodes", codes);
        return this;
    }

    setIssueSubTypeCodes(codes: Array<IssueSubType>): this {
        this.set("issueSubTypeCodes", codes);
        return this;
    }

    setIssueTypeCodes(codes: Array<IssueType>): this {
        this.set("issueTypeCodes", codes);
        return this;
    }

    setIssueWatchingStateCode(code: IssueWatchingState): this {
        this.set("issueWatchingStateCode", code);
        return this;
    }

    setReportedByUserUid(uid: string): this {
        this.set("reportedByUserUid", uid);
        return this;
    }

    setAssigneeUserUid(uid: string): this {
        this.set("assigneeUserUid", uid);
        return this;
    }

    setIssueNumbers(numbers: Array<number>): this {
        if (numbers && numbers.length > 1000) {
            throw new SmartlingException("Issue numbers array must not exceed 1000 items");
        }
        this.set("issueNumbers", numbers);
        return this;
    }

    setJobFilter(jobFilter: JobFilterDto): this {
        if (jobFilter && jobFilter.jobUids && jobFilter.jobUids.length > 1000) {
            throw new SmartlingException("Job filter jobUids array must not exceed 1000 items");
        }
        this.set("jobFilter", jobFilter);
        return this;
    }

    setStringFilter(stringFilter: StringFilterDto): this {
        this.set("stringFilter", stringFilter);
        return this;
```

**Review by vsolovei-smartling — CHANGES_REQUESTED** [review]

File: api/issues/params/base-issues-filter-parameters.ts (hunk 1 above)
Comment (PavelLoparev) [reply api/issues/params/base-issues-filter-parameters.ts]: Thanks for the suggestion — agreed that polymorphic `this` is the cleaner pattern in isolation, but I'd like to push back on applying it just here.

This PR is intentionally matching the convention already established in this codebase by two existing F-bounded base classes that predate this work:

- `BaseTagsStringsParameters<T extends BaseTagsStringsParameters<T>>` — `api/tags/params/base-tags-strings-parameters.ts:5`
- `CreateUpdateJobBaseParameters<T extends CreateUpdateJobBaseParameters<T>>` — `api/jobs/params/create-update-job-base-parameters.ts:10`

Switching `BaseIssuesFilterParameters` alone would leave 2 of 3 generic-builder bases on the F-bounded pattern and 1 on `this`, which I think is strictly worse than either uniform option — readers grepping for the pattern would have to remember which file uses which.

I'd rather keep this PR consistent with the existing convention and address the simpler-pattern question as a follow-up that converts all three classes (tags + jobs + issues) in one dedicated refactor PR. That way the change can be evaluated on its own merits as a style/convention update rather than mixed into the Issues feature work. Happy to file that follow-up.

If you'd prefer to do it all in this PR instead, say the word and I'll add tags + jobs to the diff.

File: docs/superpowers/plans/2026-05-12-issues-api-client.md
Comment (dimitrystd) [inline docs/superpowers/plans/2026-05-12-issues-api-client.md:1]: Nitpick. I'm against adding files of this type to repositories. It is just of 2k of lines already outdated code. I added to the `.gitignore` in TMS https://github.com/Smartling/tms/blob/7f260fc2548708ff2c4fadafd35797b84a306b7e/.gitignore#L27-L30

**Review by dimitrystd — APPROVED** [review]

File: api/issues/params/base-issues-filter-parameters.ts (hunk 1 above)
Comment (PavelLoparev) [reply api/issues/params/base-issues-filter-parameters.ts]: Correcting myself — I overstated the case for F-bounded as "the convention". Looking again, the codebase is split roughly 50/50:

**F-bounded base classes:**
- `BaseTagsStringsParameters` — `api/tags/params/base-tags-strings-parameters.ts`
- `CreateUpdateJobBaseParameters` — `api/jobs/params/create-update-job-base-parameters.ts`

**Polymorphic `this` base classes (with subclasses already):**
- `BaseParametersWithCallback` — `api/file-translations/params/base-parameters-with-callback.ts` (subclassed by `LanguageDetectionParameters`, `TranslateFileParameters`)
- `DownloadFileBaseParameters` — `api/files/params/download-file-base-parameters.ts` (subclassed by `DownloadFileParameters`, `DownloadMultipleFilesTranslationsParameters`)

So your suggested pattern is not a new style — there's existing precedent for it. Accepting the suggestion. I'll push a commit on this branch that converts `BaseIssuesFilterParameters` to use `this` and drops the casts in the four concrete subclasses.

Happy to also file a separate follow-up to unify tags + jobs onto the `this` pattern if you'd like the codebase fully aligned.

File: docs/superpowers/plans/2026-05-12-issues-api-client.md
Comment (PavelLoparev) [reply docs/superpowers/plans/2026-05-12-issues-api-client.md:1]: removed whole docs folder. We indeed do not need that

File: api/issues/params/base-issues-filter-parameters.ts (hunk 1 above)
Comment (PavelLoparev) [reply api/issues/params/base-issues-filter-parameters.ts]: fixed, thanks

**Review by vsolovei-smartling — APPROVED** [review]
